app/core/schedule_service.py

```
@@ -0,0 +1,434 @@
+            return ScheduleItem(
+                start_time=start_time.strftime("%H:%M"),
+                end_time=end_time.strftime("%H:%M"),
+                activity=schedule_data["activity"],
+                status=AgentStatus(status),
+                description=description,
+                location=location,
+            )
+
+        except json.JSONDecodeError as e:
+            logger.error(f"Failed to parse JSON: {e}")
+            raise ValueError(f"Invalid JSON format: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)